worldengine/cli/main.py

```diff
@@ -407,12 +407,19 @@ def main():
 
     generation_operation = (operation == 'world') or (operation == 'plates')
 
-    produce_grayscale_heightmap = args.grayscale_heightmap
-    if produce_grayscale_heightmap and not generation_operation:
+    if args.grayscale_heightmap and args.grayscale_heightmap.rfind('.') == -1:
+        print("Invalid filename `{}`, you must specify a valid image extension.".format(
+            args.grayscale_heightmap))
+        sys.exit(-1)
```

Comment:
Honestly... I want to get back into the splitting of client code and library code. The library will raise an error, and it is the job of the client code to catch, display the error and just exit.

Python doesn't have constants, but I know what you mean. Right now, it is good enough to give a non zero back that something bad happened.